drivers/watchdog/wdt_dw.c

```diff
@@ -88,8 +100,13 @@ static int dw_wdt_install_timeout(const struct device *dev, const struct wdt_tim
 	dev_data->callback = config->callback;
 #endif
 
+#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(clocks)
+	return dw_wdt_calc_period((uint32_t)reg_base, dev_data->clk_freq, config,
+				  &dev_data->config);
+#else
 	return dw_wdt_calc_period((uint32_t)reg_base, dev_config->clk_freq, config,
 				  &dev_data->config);
+#endif
```

Comment:
I find it confusing to get the clock frequency either as a hard-coded value from devicetree or via `clock_control`. Can we not just enforce using `clock_control` and get rid of all these `#ifdef`s?

Reply:
@martinjaeger, it is possible that for few platform there is no option for clock control and for some platform clock rate can ge queried through clock controller. I feel these both the methods are required.

@softwarecki, I can do that, assign a variable with clock rate from dev_data or dev_config, but still we will have a #if check.

Reply:
Why not implement clock control where it is missing?

Reply:
@martinjaeger, I think I did not understand this comment fully. Do you mean to remove the dependecy of driver to get clock frequecy from the dtsi for all the platforms? Could you please elaborate on this?

Reply:
Ideally, you would implement the clock_control driver for the missing platforms in a separate PR, the that merged and afterwards rebase this one.

Your above comment was very vague, that "it is possible that for few platform there is no option for clock control". How many and which platforms are there, actually?

As a compromise I suggest to use `dev_data->clk_freq` all the time (even if the frequency is determined from devicetree at compile-time). This would avoid the `#ifdef`s.

Reply:
@pbalsundar Yes, #ifdef will remain, but it will not contain the entire function call, which will be more readable.

Reply:
@martinjaeger @softwarecki I have added the clk_freq to the dev_data instead of dev_cfg structure.